sale_coupon_advanced_option/models/sale_coupon_program.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+
+
+class SaleCouponProgram(models.Model):
+    _inherit = "sale.coupon.program"
+
+    # Add possibility to use discount only on first order of a customer
+    first_order_only = fields.Boolean(
+        string="Apply only first",
+        default=False,
+        help="Apply only on the first order of each client matching the conditions",
+    )
+
+    def _check_promo_code(self, order, coupon_code):
+        if (
+            self.first_order_only
+            and len(
+                self.env["sale.order"].search(
+                    [
+                        ("partner_id", "=", order.partner_id.id),
+                        ("state", "!=", "cancel"),
```

Comment:
Shouldn't we exclude quotation state also? `"state", "not int" ("cancel", "draft")`

Reply:
@santostelmo  good point but we discussed with @gurneyalex that we are on safe side when considering draft so - for example customer started to create order but didn't finish, order will be saved as a draft and can be recovered and finished at any time.

Reply:
I understand rule, if there is `draft` orders pending to be validated the customer can not benefit of the coupon.